Source/WTF/wtf/text/WTFString.h

```diff
@@ -58,14 +58,16 @@ class String final {
 
     // Construct a string with UTF-16 data.
     WTF_EXPORT_PRIVATE String(const UChar* characters, unsigned length);
-    ALWAYS_INLINE String(Span<const UChar> characters) : String(characters.data(), characters.size()) { }
 
     // Construct a string with Latin-1 data.
     WTF_EXPORT_PRIVATE String(const LChar* characters, unsigned length);
     WTF_EXPORT_PRIVATE String(const char* characters, unsigned length);
```

Comment:
If I understand the String design policy correctly, we will have to convert these 8bit constructors to `fromLatin1`.
> fromLatin1(const LChar* characters, unsigned length);
> fromLatin1(const char* characters, unsigned length);
If so, I think `fromLatin` should be able to accept `Span<const LChar>`
> fromLatin(Span<const LChar> span)

Reply:
Maybe, but then we could have the same ambiguity problem again: if you call `String::fromLatin1` on a literal it could use the length of the array rather than the length of the literal. Just depends on how likely we think that mistake is. Requiring the word "span" might help.

Reply:
Yes, the `String(const CharType*, unsigned)` constructor parameters are effectively spans, so they could all be converted into named constructors taking in Span objects. It's a hefty job, not impossible. Loss of the `unsigned` type for length wouldn't be great, but I think it could be implemented properly.

Reply:
We have `fromLatin1(const char*)`. `String::fromLatin1("string")` calls it. No ambiguity problem in this case.

Reply:
I attached a patch for the discussion. https://bugs.webkit.org/show_bug.cgi?id=251385#c2